src/sources/host_metrics/network.rs

```diff
@@ -44,42 +44,43 @@ impl HostMetrics {
                     .await
                 {
                     let interface = counter.interface();
+                    let tags = metric_tags!("device" => interface);
                     output.counter(
                         "network_receive_bytes_total",
                         counter.bytes_recv().get::<byte>() as f64,
-                        MetricTags::from([(String::from("device"), interface.to_string())]),
+                        tags.clone(),
                     );
                     output.counter(
                         "network_receive_errs_total",
                         counter.errors_recv() as f64,
-                        MetricTags::from([(String::from("device"), interface.to_string())]),
+                        tags.clone(),
                     );
                     output.counter(
                         "network_receive_packets_total",
                         counter.packets_recv() as f64,
-                        MetricTags::from([(String::from("device"), interface.to_string())]),
+                        tags.clone(),
                     );
                     output.counter(
                         "network_transmit_bytes_total",
                         counter.bytes_sent().get::<byte>() as f64,
-                        MetricTags::from([(String::from("device"), interface.to_string())]),
-                    );
-                    output.counter(
-                        "network_transmit_errs_total",
-                        counter.errors_sent() as f64,
-                        MetricTags::from([(String::from("device"), interface.to_string())]),
+                        tags.clone(),
                     );
                     #[cfg(any(target_os = "linux", windows))]
                     output.counter(
                         "network_transmit_packets_drop_total",
                         counter.drop_sent() as f64,
-                        MetricTags::from([(String::from("device"), interface.to_string())]),
+                        tags.clone(),
                     );
                     #[cfg(any(target_os = "linux", windows))]
                     output.counter(
                         "network_transmit_packets_total",
                         counter.packets_sent() as f64,
-                        MetricTags::from([(String::from("device"), interface.to_string())]),
+                        tags.clone(),
+                    );
+                    output.counter(
+                        "network_transmit_errs_total",
```

Comment:
Curious why you moved this to the bottom? It was alphabetical, and the `recv` side still is.

Reply:
That lets the last `tags` avoid the `.clone()`. The last use of value cannot be cloned due to a clippy check, and the last two metrics are optional without reordering, causing issues.

Reply:
Ah, 👍 I missed that and just thought it was an arbitrary reordering - like errors last or something 😄